backend/src/main/kotlin/outcobra/server/validator/SchoolYearValidator.kt

```diff
@@ -0,0 +1,24 @@
+package outcobra.server.validator
+
+import org.springframework.stereotype.Component
+import outcobra.server.model.QSchoolYear
+import outcobra.server.model.SchoolYear
+import outcobra.server.model.repository.SchoolYearRepository
+import outcobra.server.util.DateOverlabUtil
+import javax.inject.Inject
+
+@Component
+open class SchoolYearValidator
+@Inject constructor(val schoolYearRepository: SchoolYearRepository) {
+
+    fun validateSchoolYearCreation(schoolYear: SchoolYear): Boolean {
+
+        val predicate = QSchoolYear.schoolYear.schoolClass.id.eq(schoolYear.schoolClass.id)
+        val schoolYears = schoolYearRepository.findAll(predicate).toList()
+        return schoolYears.all { checkOverlap(it, schoolYear) }
+    }
+
```

Comment:
Have you checked updating? pretty sure the updated schoolYear will overlap with itself